test/tests_for_sdkmr/ee2_SDKMethodRunner_test_EE2Runjob_test.py

```diff
@@ -541,6 +544,159 @@ def test_retry_job_with_params_and_nci_and_src_ws_objs(self, rq_mock, condor_moc
         # TODO Retry a job without an app_id
         # TODO Check narrative_cell_info
 
+    def get_batch_jobs(self):
+        job = get_example_job_as_dict(
+            user=self.user_id, wsid=None, source_ws_objects=[]
+        )
+        job2 = get_example_job_as_dict(
+            user=self.user_id, wsid=None, source_ws_objects=[]
+        )
+        job3 = get_example_job_as_dict(
+            user=self.user_id, wsid=None, source_ws_objects=[]
+        )
+        jobs = [job, job2, job3]
+        return jobs
+
+    @requests_mock.Mocker()
+    @patch("lib.execution_engine2.utils.Condor.Condor", autospec=True)
+    def test_retry_batch_and_cancel_batch(self, rq_mock, condor_mock):
+        rq_mock.add_matcher(
+            run_job_adapter(
+                ws_perms_info={"user_id": self.user_id, "ws_perms": {self.ws_id: "a"}}
+            )
+        )
+        runner = self.getRunner()
+        runner.get_condor = MagicMock(return_value=condor_mock)
+        jobs = self.get_batch_jobs()
+        condor_mock.run_job = MagicMock(
+            return_value=SubmissionInfo(clusterid="test", submit=jobs[0], error=None)
+        )
+        job_ids = runner.run_job_batch(
+            params=copy.deepcopy(jobs), batch_params={"wsid": self.ws_id}
+        )
+        batch_job_id = job_ids["batch_id"]
+        child_jobs = job_ids["child_job_ids"]
+
+        # TEST OUT ERROR CONDITIONS
+        for status_list in [
+            None,
+            [],
+        ]:
```

Comment:
I think it'd be easier to understand if you split out all these tests into separate functions -- e.g. `test_retry_batch__no_batch_id`, `test_retry_batch__no_status_list`, `test_retry_batch__invalid_status_list`, `test_retry_batch__invalid_batch_id`, `test_retry_batch__no_jobs_with_status`, etc., etc. Stepping through the functions and ensuring there are clearly-named tests in all the appropriate places will make it a lot easier to track down issues.